diff --git a/node_modules/react-native-view-shot/android/src/main/java/fr/greweb/reactnativeviewshot/RNViewShotModule.java b/node_modules/react-native-view-shot/android/src/main/java/fr/greweb/reactnativeviewshot/RNViewShotModule.java
index 3bf6b6f..30593b0 100644
--- a/node_modules/react-native-view-shot/android/src/main/java/fr/greweb/reactnativeviewshot/RNViewShotModule.java
+++ b/node_modules/react-native-view-shot/android/src/main/java/fr/greweb/reactnativeviewshot/RNViewShotModule.java
@@ -84,12 +84,13 @@ public class RNViewShotModule extends ReactContextBaseJavaModule {
         final Integer scaleWidth = options.hasKey("width") ? (int) (dm.density * options.getDouble("width")) : null;
         final Integer scaleHeight = options.hasKey("height") ? (int) (dm.density * options.getDouble("height")) : null;
         final String resultStreamFormat = options.getString("result");
+        final String fileName = options.hasKey("fileName") ? options.getString("fileName") : null;
         final Boolean snapshotContentContainer = options.getBoolean("snapshotContentContainer");
 
         try {
             File outputFile = null;
             if (Results.TEMP_FILE.equals(resultStreamFormat)) {
-                outputFile = createTempFile(getReactApplicationContext(), extension);
+                outputFile = createTempFile(getReactApplicationContext(), extension, fileName);
             }
 
             final Activity activity = getCurrentActivity();
@@ -163,7 +164,7 @@ public class RNViewShotModule extends ReactContextBaseJavaModule {
      * whichever is available and has more free space.
      */
     @NonNull
-    private File createTempFile(@NonNull final Context context, @NonNull final String ext) throws IOException {
+    private File createTempFile(@NonNull final Context context, @NonNull final String ext, String fileName) throws IOException {
         final File externalCacheDir = context.getExternalCacheDir();
         final File internalCacheDir = context.getCacheDir();
         final File cacheDir;
@@ -182,6 +183,9 @@ public class RNViewShotModule extends ReactContextBaseJavaModule {
         }
 
         final String suffix = "." + ext;
+        if (fileName != null) {
+            return File.createTempFile(fileName, suffix, cacheDir);
+        }
         return File.createTempFile(TEMP_FILE_PREFIX, suffix, cacheDir);
     }
 }
diff --git a/node_modules/react-native-view-shot/src/index.d.ts b/node_modules/react-native-view-shot/src/index.d.ts
index 00f36ad..73abf03 100644
--- a/node_modules/react-native-view-shot/src/index.d.ts
+++ b/node_modules/react-native-view-shot/src/index.d.ts
@@ -13,6 +13,10 @@ declare module 'react-native-view-shot' {
     import { ViewStyle } from 'react-native'
 
     export interface CaptureOptions {
+        /**
+         * (Android only) the file name of the file. Must be at least 3 characters long.
+         */
+        fileName?: string;
         /**
          * (number): the width and height of the final image (resized from the View bound. don't provide it if you want
          * the original pixel size).
